sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java

```diff
@@ -481,6 +482,10 @@ public CosmosContainerProperties createContainerIfNotExists(CosmosEntityInformat
                     new CosmosContainerProperties(information.getContainerName(), information.getPartitionKeyPath());
                 cosmosContainerProperties.setDefaultTimeToLiveInSeconds(information.getTimeToLive());
                 cosmosContainerProperties.setIndexingPolicy(information.getIndexingPolicy());
+                final UniqueKeyPolicy uniqueKeyPolicy = information.getUniqueKeyPolicy();
+                if (uniqueKeyPolicy != null) {
+                    cosmosContainerProperties.setUniqueKeyPolicy(uniqueKeyPolicy);
```

Comment:
what if customer already defined some uniqueKey policy on the container? Will this override what already defined?

Reply:
If the customer already defined uniqueKey policy, then they won't use any annotation. 
Annotation is supposed to take precedence on their existing unique key policy. That's why the null check here :)

Reply:
If you see this PR - https://github.com/Azure/azure-sdk-for-java/pull/22246
which fixed the indexing policy override. Same intention here to add null check and only apply the indexing policy if it is not null. 
If you see `CosmosEntityInformation`, I am returning null if unique key policy is not specified or is empty.

Reply:
sounds good, thanks for the explanation 👍